src/main/java/com/backbase/ct/bbfuel/setup/CapabilitiesDataSetup.java

```diff
@@ -179,12 +188,34 @@ private void ingestActionsPerUser() {
 
     private void ingestBillPayUsers() {
         if (this.globalProperties.getBoolean(PROPERTY_INGEST_BILLPAY)) {
-            this.accessControlSetup.getLegalEntitiesWithUsersExcludingSupport()
-                .stream()
-                .map(LegalEntityWithUsers::getUserExternalIds)
-                .flatMap(List::stream)
-                .collect(Collectors.toList())
-                .forEach(this.billpayConfigurator::ingestBillPayUser);
+            this.accessControlSetup.getLegalEntitiesWithUsersExcludingSupport().forEach((le) -> {
+
+                List<Account> accounts = new ArrayList<>();
+                String externalUserId = le.getUserExternalIds().get(0);
```

Comment:
Would be better to add another `if (ingest.billpay.accounts=true)` around this logic so that the arrangement details are only retrieved if necessary.